src/sys/socket/multicast.rs

```diff
@@ -37,4 +37,10 @@ impl ipv6_mreq {
             ipv6mr_interface: 0,
         }
     }
+    pub fn new(group: Ipv6Addr, interface: c_uint) -> ipv6_mreq {
```

Comment:
Yeah, overloading by argument count doesn't work, as evident from CI failure:

```
src/sys/socket/multicast.rs:40:5: 45:6 error: duplicate associated function [E0201]
```

Given that these aren't actually system functions, I think we'd be best off if *both* had more representative names. I also think there's work to be done here with moving to newtypes around the libc types..

My preference is to defer fixing #323 until we can get libc types + newtypes sorted out here.